apps/web/src/components/layout/components/v2/SyncInfoModalTrigger.tsx

```diff
@@ -1,11 +1,48 @@
-import { Button } from '@novu/novui';
+import React, { useMemo, useState } from 'react';
+import { useQuery } from '@tanstack/react-query';
+import CryptoJS from 'crypto-js';
+import styled from '@emotion/styled';
+
+import { colors, Tooltip, useColorScheme } from '@novu/design-system';
+import { Button, Text } from '@novu/novui';
 import { IconOutlineCloudUpload } from '@novu/novui/icons';
-import { useState } from 'react';
+import { INotificationTemplate } from '@novu/shared';
+
 import { useTelemetry } from '../../../../hooks/useNovuAPI';
 import { SyncInfoModal } from './SyncInfoModal';
+import { useDiscover } from '../../../../studio/hooks';
+import { getNotificationsList } from '../../../../api/notification-templates';
+import { When } from '../../../utils/When';
 
 export function SyncInfoModalTrigger() {
   const [showSyncInfoModal, setShowSyncInfoModal] = useState(false);
+  const { data: bridgeDiscoverData, isLoading: isLoadingBridgeWorkflows } = useDiscover();
+  const {
+    data: originData,
+    isLoading: isLoadingOriginWorkflows,
+    refetch,
+  } = useQuery(
+    ['origin-workflows'],
+    async () => {
+      return getNotificationsList({ page: 0, limit: 100 });
+    },
+    {}
+  );
+
+  const isSynced = useMemo(() => {
+    if (isLoadingBridgeWorkflows || isLoadingOriginWorkflows) {
+      return true;
+    }
+
+    const bridgeDiscoverWorkflows = bridgeDiscoverData?.workflows || undefined;
+    const originWorkflows = originData?.data.map((workflow: INotificationTemplate) => workflow.rawData) || undefined;
+
+    const bridgeDiscoverWorkflowsHash = createHash(bridgeDiscoverWorkflows);
+    const storedWorkflowsHash = createHash(originWorkflows);
+
+    return storedWorkflowsHash === bridgeDiscoverWorkflowsHash;
+  }, [bridgeDiscoverData, originData, isLoadingBridgeWorkflows, isLoadingOriginWorkflows]);
```

Comment:
<!-- Chores are simple tasks that must be done before the subject can be “officially” accepted. Usually, these comments reference some common process. Try to leave a link to the process description so that the reader knows how to resolve the chore.  -->
☑ **chore:** Please extract to its own hook to avoid overloading this component‏

Reply:
done